tracker/gioui/preferences.go

```diff
@@ -0,0 +1,65 @@
+package gioui
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+
+	"gioui.org/unit"
+)
+
+type (
+	Preferences struct {
+		Window   WindowPreferences
+		YmlError error
+	}
+
+	WindowPreferences struct {
+		Width     int
+		Height    int
+		Maximized bool `yaml:",omitempty"`
+	}
+)
+
+//go:embed preferences.yml
+var defaultPreferencesYaml []byte
+
+func loadDefaultPreferences() Preferences {
+	var preferences Preferences
+	err := yaml.Unmarshal(defaultPreferencesYaml, &preferences)
+	if err != nil {
+		panic(fmt.Errorf("failed to unmarshal preferences: %w", err))
+	}
+	return preferences
+}
+
+// ReadCustomConfigYml modifies the target argument, i.e. needs a pointer
+func ReadCustomConfigYml(filename string, target interface{}) (exists bool, err error) {
+	configDir, err := os.UserConfigDir()
+	if err != nil {
+		return false, err
+	}
+	path := filepath.Join(configDir, "sointu", filename)
+	bytes, err2 := os.ReadFile(path)
+	if err2 != nil {
+		return false, err2
+	}
+	err = yaml.Unmarshal(bytes, target)
+	return true, err
+}
+
+func NewPreferences() Preferences {
```

Comment:
nit: I prefer reserving word "New" only for heap allocations, so typically functions returning pointers to structs. This matches the meaning of "new" in C++.  Using `MakePreferences` would match how `new` and `make` keywords work in go: `new(type)` returns a pointer to a type, whereas `make(type)` returns type

Reply:
Sometimes dropping the Make altogether and using some convenient short word is great. For example, gioui has layout.Rigid basically just doing similar thing i.e. constructing a struct, but it would be pretty verbose to write "layout.MakeFlexChild" every time. But here, the MakePreferences won't be called too many times, so I think a bit more lengthy name is fine.

Reply:
sure, I didn't notice before that the New...() only return pointers. but I have no feelings whatsoever about that name for now, so... yeah :)